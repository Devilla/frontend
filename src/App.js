import React from 'react';
import { browserHistory } from 'react-router';
import { AppContainer } from 'react-hot-loader';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import rootSaga from './sagas';
import reducer from './ducks';
import thunk from 'redux-thunk';
import { syncHistoryWithStore } from 'react-router-redux';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import createSagaMiddleware from 'redux-saga';
import { StripeProvider } from 'react-stripe-elements';

const sagaMiddleware = createSagaMiddleware();

export const store = createStore(reducer, composeWithDevTools(
  applyMiddleware(sagaMiddleware, thunk)
));

const routerHistory = syncHistoryWithStore(browserHistory, store, {selectLocationState: state => state.get('router')});


sagaMiddleware.run(rootSaga);

const App = ({Component}) =>
  <AppContainer errorReporter={Error}>
    <Provider store={store}>
      <StripeProvider apiKey={process.env.NODE_ENV === 'production' ? process.env.REACT_APP_STRIPE_KEY : process.env.REACT_APP_DEVELOPMENT_STRIPE_KEY}>
        <Component routerHistory={routerHistory} />
      </StripeProvider>
    </Provider>
  </AppContainer>;

export default App;
