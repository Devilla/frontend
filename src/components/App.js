import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import  '../assets/stylesheets/slick.css'
import  '../assets/stylesheets/slick-theme.css'
import  '../assets/stylesheets/style.css'

import Header from './_common/Header';
import Footer from './_common/Footer';
import { Spinner } from './index.js';

class App extends Component {
  render(){
    return (
        <div className="app-container">
          <Header />

          <Spinner loading={this.props.loading} />
          <div className="content">
            {this.props.children}
          </div>
          <Footer />
        </div>
    );
  }
};

App.propTypes = {
  children: PropTypes.node
};

const mapStateToProps = state => ({
  loading: state.getIn(['loading', 'state'])
});

export default connect(mapStateToProps)(App);
