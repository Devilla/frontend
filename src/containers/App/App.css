@import url('https://fonts.googleapis.com/css?family=Open+Sans:200,300,400,400i,500,600,700');
@import url('https://fonts.googleapis.com/css?family=Source+Sans+Pro:200,300,400,400i,600,700');

body {
  font-family: "Raleway, Source Sans Pro, Verdana, Helvetica", sans-serif;
}
