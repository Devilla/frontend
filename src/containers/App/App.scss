@import url('https://fonts.googleapis.com/css?family=Open+Sans:200,300,400,400i,500,600,700');
@import url('https://fonts.googleapis.com/css?family=Source+Sans+Pro:200,300,400,400i,600,700');

body {
  font-family: "Raleway, Source Sans Pro, Verdana, Helvetica", sans-serif;
  display: block;
  margin: 0px;
}

.website-app {
  .alert-warning {
    background-color: #f2dede;
    border-color: #ebccd1;
    color: #a94442;
  }
  .content {
    padding-top: 100px;
  }
  .toast-style {
    padding: 17px 7px;
    background-color: rgba(0, 0, 0, 0.9) !important;
    color: #fff !important;
    text-align: center !important;
    height: 10px !important;
    border-radius: 5px !important;
    button {
      height: 1.5em;
      color: #fff;
    }
  }

  .transition-wrapper {
    position: relative;
    z-index: 1;
    .transition-item {
      position: fixed;
      top: 0;
      right: 0;
      left: 0;
    }
  }
  .signin-container,
  .authpage,
  .websitepolicy-container,
  .websitehowitworks-container,
  .websitehome-container,
  .WebsiteTerms-container,
  .websitepricing-container,
  .website-payment-container,
  .websiteintegrations-container,
  .demopage-container,
  .website-cust-story,
  .websitecontact-container,
  .websiteabout-container,
  .WebsiteFeature-container,
  .affiliateregister-container
   {
    overflow: hidden;
    box-sizing: border-box;
    padding: 20px;
    min-height: 100vh;
    transition: 500ms ease-out;

    &.transition-leave {
      opacity: 1;
    }

    &.transition-leave.transition-leave-active {
      opacity: 1;
      transform: translate3d(100%, 0, 0);
    }
  }
}
