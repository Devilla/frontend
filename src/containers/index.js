export { default as App } from './App/App';
export { default as DashboardContainer } from './DashboardContainer/DashboardContainer';
export { default as ConnectPage } from './ConnectPage/ConnectPage';
export { default as AnalyticsContainer } from './AnalyticsContainer/AnalyticsContainer';
export { default as NewCampaignContainer } from './NewCampaignContainer/NewCampaignContainer';
export { default as OauthContainer } from './OauthContainer/OauthContainer';
