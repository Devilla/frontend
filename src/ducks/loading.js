import { Map } from 'immutable';

const action = name => `zikher/modals/${name}`;
const LOAD = action('LOAD');
const LOADED = action('LOADED');

export const load = () => ({ type: LOAD });
export const loaded = () => ({ type: LOADED });

const initialState = Map({
  state: false,
});

const loading = (state = initialState, action) => {
  switch (action.type) {
    case LOAD:
      return state.set('state', true);

    case LOADED:
      return state.set('state', false);

    default:
      return state;
  }
};

export default loading;
