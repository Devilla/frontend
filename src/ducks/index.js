import { combineReducers } from 'redux-immutable';
import { routerReducer } from 'react-router-redux'

import profile from './profile';
import auth from './auth';
import plan from './plan';
import payment from './payment';
import notification from './notification';
import campaign from './campaign';
import rules from './rules';
import loading from './loading';
import configuration from './configuration';
import elastic from './elastic';
import pageurl from './pageurl';
import webhooks from './webhooks';
import subcampaign from './subcampaign';

const reducer = combineReducers({
  profile,
  auth,
  plan,
  payment,
  notification,
  campaign,
  rules,
  loading,
  configuration,
  elastic,
  pageurl,
  webhooks,
  subcampaign,
  router: routerReducer
});

export default reducer;
