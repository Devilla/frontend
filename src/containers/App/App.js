import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { ToastContainer } from 'react-toastify';
import { browserHistory } from 'react-router';
import { connect } from 'react-redux';
import { Spinner } from '../../components/index.js';
import { WebsiteHeader, WebsiteFooter } from 'components';

import { checkTokenExists } from 'ducks/auth';

import './scss/stack-interface.scss';
import './scss/socicon.css';
import './scss/iconsmind.scss';
import './scss/bootstrap.scss';
import './scss/theme.scss';
import './scss/custom.scss';
import './scss/font-raleway.scss';
import './scss/flickity.scss';
import './scss/font-sourcesanspro.scss';
import './App.scss';

class App extends Component {
  componentWillMount() {
    this.checkLogin((err) => {
      if(!err) {
        browserHistory.push('/dashboard');
      }
    });
  }

  checkLogin(callback) {
    const cookie = localStorage.getItem('authToken');
    const authToken = cookie
      ? JSON.parse(cookie)
      : null;
    if (authToken) {
      this.props.checkTokenExists(authToken);
      callback()
    } else {
      callback("not logged in")
    }
  }

  render(){
    return (
      <div className="website-app">
        <div className="basic-gradient-light" data-smooth-scroll-offset="77">
           <WebsiteHeader />
           <Spinner loading={this.props.loading} />
           <div className="content">
             {this.props.children}
           </div>
           <WebsiteFooter />
        </div>
        <ToastContainer hideProgressBar={true} />
      </div>
    );
  }

}

const mapDispatchToProps = {
  checkTokenExists
};

export default connect(null, mapDispatchToProps)(App);
