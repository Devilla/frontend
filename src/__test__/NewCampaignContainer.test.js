import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import {createStore} from 'redux'
import { load, loaded } from 'ducks/loading';
import { NewCampaignContainer } from 'containers';
import { store } from 'App';

// Snapshot for Home React Component
describe('>>> NewCampaignContainer --- Snapshot',()=>{
    it('+++capturing Snapshot of NewCampaignContainer', () => {
        const renderedValue =  renderer.create(<NewCampaignContainer store={store} />).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });
});
