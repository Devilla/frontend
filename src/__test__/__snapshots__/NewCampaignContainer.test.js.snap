// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> NewCampaignContainer --- Snapshot +++capturing Snapshot of NewCampaignContainer 1`] = `
<div
  className="NewCampaignContainer"
>
  <button
    className="btn btn-outline-primary goliveRight waves-light waves-effect number"
    data-target="#myModallive"
    data-toggle="modal"
    onClick={[Function]}
    type="button"
  >
    <i
      className="fi-location"
    />
     Go Live
  </button>
  <div
    className="content fill campaign-container"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-md-12"
        >
          <div
            className="card-box"
          >
            <h4
              className="header-title"
            >
              Create Your Campaign
            </h4>
            <hr />
            <form
              onSubmit={[Function]}
            >
              <div
                className="row"
              >
                <div
                  className="col-md-6"
                >
                  <div
                    className="form-group"
                  >
                    <label
                      className="text-muted h6 control-label"
                    >
                      Name
                    </label>
                    <input
                      id="campaignname"
                      name="campaignname"
                      onChange={[Function]}
                      placeholder="example: Acme Co, Blog, Online Store"
                      required={true}
                      type="text"
                      value="Influence"
                    />
                    <span
                      className="help-block"
                    >
                      <p
                        className="website-error"
                      >
                        
                      </p>
                    </span>
                  </div>
                </div>
                <div
                  className="col-md-6"
                >
                  <div
                    className="form-group"
                  >
                    <label
                      className="text-muted h6 control-label"
                    >
                      Website URL
                    </label>
                    <input
                      defaultValue=""
                      id="website"
                      name="website"
                      onChange={[Function]}
                      placeholder="http://"
                      required={true}
                      type="text"
                    />
                    <span
                      className="help-block"
                    >
                      <p
                        className="website-error"
                      >
                        
                      </p>
                    </span>
                  </div>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col-md-6"
                >
                  <div
                    className="form-group"
                  >
                    <label
                      className="text-muted h6 control-label"
                    >
                      Conversions per day
                    </label>
                      
                    <i
                      className="fa fa-info-circle"
                      data-delay="{\\"show\\":\\"0\\", \\"hide\\":\\"100\\"}"
                      data-toggle="tooltip"
                      height="50px"
                      title="Fill in for better monitoring"
                      width="50px"
                    >
                       
                    </i>
                    <input
                      id="averageCustomer"
                      name="averageCustomer"
                      onChange={[Function]}
                      placeholder="Number of customer Signups"
                      required={true}
                      type="number"
                      value=""
                    />
                    <span
                      className="help-block"
                    >
                      <p
                        className="website-error"
                      >
                        
                      </p>
                    </span>
                  </div>
                </div>
              </div>
              <button
                className="btn btn-primary waves-light waves-effect newcamp-btn ml-2 pl-4 pr-4"
                type="submit"
              >
                 Create Your Campaign 
              </button>
              <div
                className="clearfix"
              />
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="modal fade show-modal"
    id="myModallive"
    role="dialog"
  >
    <div
      className="modal-dialog"
    >
      <div
        className="modal-content align-modal"
      >
        <div
          className="modal-header"
        >
          <h4
            className="modal-title"
          >
            
          </h4>
        </div>
        <div
          className="modal-body"
        >
          
        </div>
        <div
          className="modal-footer"
        >
          <button
            className="btn btn-primary close-btn"
            data-dismiss="modal"
            onClick={[Function]}
            type="button"
          >
            
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
